bindata/worker-osp/007-worker-osp-machineset.yaml

```diff
@@ -24,7 +24,9 @@ spec:
     spec:
       providerSpec:
         value:
-          hostSelector: {}
+          hostSelector:
+            matchLabels:
+              ospRole: {{ .WorkerOspRole }}
```

Comment:
Just keep in mind that, without the associated label on the BMH resource, this change by itself will prevent the operator from being able to scale the machineset.

Reply:
yes thats correct. there must be always enough nodes labeled for the role